### ISSUE TITLE
Installation does not work (channel-independent issue)

### ISSUE THREAD

**Issue by justinshaffer** [biocore/q2-fragment-insertion#48]
Hello,

I've run into an error when installing this package into a QIIME2 (2018.8) conda environment (Miniconda3-latest-Linux-x86_64) installed into my home directory on a computing cluster (i.e., barnacle). 

This is the code I ran to install the package:

$ conda install -c anaconda -c defaults -c conda-forge -c bioconda -c https://conda.anaconda.org/biocore q2-fragment-insertion

This is the error that prints to screen when running the install code:

# BEGINNING OF ERROR PRINTED TO SCREEN
Solving environment: failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/exceptions.py", line 819, in __call__
        return func(*args, **kwargs)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/cli/main.py", line 78, in _main
        exit_code = do_call(args, p)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 77, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/cli/main_install.py", line 11, in execute
        install(args, parser, 'install')
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/cli/install.py", line 235, in install
        force_reinstall=context.force,
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 518, in solve_for_transaction
        force_remove, force_reinstall)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 451, in solve_for_diff
        final_precs = self.solve_final_state(deps_modifier, prune, ignore_pinned, force_remove)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 180, in solve_final_state
        index, r = self._prepare(prepared_specs)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 592, in _prepare
        self.subdirs, prepared_specs)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/index.py", line 215, in get_reduced_index
        new_records = query_all(spec)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/index.py", line 184, in query_all
        return tuple(concat(future.result() for future in as_completed(futures)))
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/subdir_data.py", line 95, in query
        self.load()
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/subdir_data.py", line 149, in load
        _internal_state = self._load()
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/subdir_data.py", line 246, in _load
        _internal_state = self._process_raw_repodata_str(raw_repodata_str)
      File "/home/jpshaffer/software/miniconda3/lib/python3.7/site-packages/conda/core/subdir_data.py", line 369, in _process_raw_repodata_str
        info['fn'] = fn
    TypeError: 'NoneType' object does not support item assignment

`$ /home/jpshaffer/software/miniconda3/bin/conda install -c anaconda -c defaults -c conda-forge -c bioconda -c https://conda.anaconda.org/biocore q2-fragment-insertion`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=qiime2-2018.8
                CONDA_EXE=/home/jpshaffer/software/miniconda3/bin/conda
             CONDA_PREFIX=/home/jpshaffer/software/miniconda3/envs/qiime2-2018.8
    CONDA_PROMPT_MODIFIER=(qiime2-2018.8)
         CONDA_PYTHON_EXE=/home/jpshaffer/software/miniconda3/bin/python
               CONDA_ROOT=/home/jpshaffer/software/miniconda3
              CONDA_SHLVL=1
                  MANPATH=/opt/slurm-18.08.0/share/man:/opt/torque-4.2.8/man:
          MODULEPATH=/opt/modules/Modules/versions:/opt/modules/Modules/$MODULE_VERSION/mod
                          ulefiles:/opt/modules/Modules/modulefiles
                     PATH=/home/jpshaffer/software/miniconda3/envs/qiime2-2018.8/bin:/home/jpsha
                          ffer/software/miniconda3/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/bin
                          :/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/opt/gold/2.2.0.5/sbin:/opt/
                          gold/2.2.0.5/bin:/opt/torque-4.2.8/bin:/opt/torque-4.2.8/sbin:/opt/mau
                          i-3.3.1/bin:/opt/slurm-18.08.0/bin:/opt/slurm-18.08.0/sbin
         PYTHONNOUSERSITE=/home/jpshaffer/software/miniconda3/envs/qiime2-2018.8/lib/python*/sit
                          e-packages/
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : qiime2-2018.8
    active env location : /home/jpshaffer/software/miniconda3/envs/qiime2-2018.8
            shell level : 1
       user config file : /home/jpshaffer/.condarc
 populated config files : 
          conda version : 4.5.11
    conda-build version : not installed
         python version : 3.7.0.final.0
       base environment : /home/jpshaffer/software/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/anaconda/linux-64
                          https://conda.anaconda.org/anaconda/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/bioconda/linux-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://conda.anaconda.org/biocore/linux-64
                          https://conda.anaconda.org/biocore/noarch
          package cache : /home/jpshaffer/software/miniconda3/pkgs
                          /home/jpshaffer/.conda/pkgs
       envs directories : /home/jpshaffer/software/miniconda3/envs
                          /home/jpshaffer/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.11 requests/2.19.1 CPython/3.7.0 Linux/2.6.32-573.26.1.el6.x86_64.debug centos/6.6 glibc/2.12
                UID:GID : 420084:550
             netrc file : None
           offline mode : False

An unexpected error has occurred. Conda has prepared the above report.
If submitted, this report will be used by core maintainers to improve
future releases of conda.

# END OF ERROR PRINTED TO SCREEN

I was able to reproduce the error after uninstalling and reinstalling both Miniconda and the QIIME2 environment.

Please let me know if you need additional information to troubleshoot this error.

Thanks in advance and best wishes,

Justin

**Comment by wasade** [comment]
Seems like this may be an upstream problem with conda, would it be possible
to open an issue on their tracker to see if it is?

**Comment by justinshaffer** [comment]
Thanks, Daniel. Yes, I will do that now.

Justin

**Comment by justinshaffer** [comment]
The issue is indeed with conda. I was able to resolve this issue by following instructions detailed in an error report for Anaconda:

https://github.com/conda/conda/issues/7870

There appears to be an issue with using the "anaconda" channel for acquiring packages. I resolved this issue by appending as channels only "https://conda.anaconda.org/biocore", "bioconda", "conda-forge", and "defaults" to my ~/.condarc file. A similar solution may also be to simply remove "anaconda" from the channel list in the install script above.

Thanks for your help,

Justin
